framework/decode/custom_vulkan_ascii_consumer.h

```diff
@@ -0,0 +1,215 @@
+
+#include "vulkan/vulkan.h"
+
+#include <sstream>
+#include <string>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+inline std::string HandleIdToString(format::HandleId handleId)
+{
+    std::stringstream strStrm;
+    if (handleId)
+    {
+        strStrm << "\"0x" << (const void*)handleId << "\"";
+    }
+    else
+    {
+        strStrm << "\"VK_NULL_HANDLE\"";
+    }
+    return strStrm.str();
+}
+
+template <typename VkHandleType>
+inline std::string HandlePointerDecoderToString(HandlePointerDecoder<VkHandleType>* pObj)
+{
+    auto pDecodedObj = pObj ? pObj->GetPointer() : nullptr;
+    return HandleIdToString(pDecodedObj ? *pDecodedObj : format::kNullHandleId);
+}
+
+template <typename VkEnumType>
+inline std::string EnumPointerDecoderToString(PointerDecoder<VkEnumType>* pObj)
+{
+    auto pDecodedObj = pObj ? pObj->GetPointer() : nullptr;
+    return pDecodedObj ? ('"' + util::ToString(*pDecodedObj) + '"') : "null";
+}
+
+template <typename T>
+inline std::string DataPointerDecoderToString(const T& pObj)
+{
+    // NOTE : Currently we just output the integer value of data void pointers, for
+    //  fields that use a PointerDecoder<> we output the address of the decoder
+    //  itself...this isn't terribly useful, but the capture time address wouldn't
+    //  be a terribly useful thing anyway.  There are structures that have data
+    //  fields that can be interpreted correctly...if we want to output interpreted
+    //  void pointer data we can implement custom handlers for those structures.
+    return "\"" + util::PtrToString(pObj) + "\"";
+}
+
+template <typename PointerDecoderType>
+inline std::string PointerDecoderToString(PointerDecoderType* pObj,
+                                          util::ToStringFlags toStringFlags = util::kToString_Default,
+                                          uint32_t            tabCount      = 0,
+                                          uint32_t            tabSize       = 4)
+{
+    auto pDecodedObj = pObj ? pObj->GetPointer() : nullptr;
+    return pDecodedObj ? util::ToString(*pDecodedObj, toStringFlags, tabCount, tabSize) : "null";
+}
+
+inline std::string DescriptorUpdateTemplateDecoderToString(const DescriptorUpdateTemplateDecoder* pObj,
+                                                           util::ToStringFlags toStringFlags = util::kToString_Default,
+                                                           uint32_t            tabCount      = 0,
+                                                           uint32_t            tabSize       = 4)
+{
+    VkWriteDescriptorSet                         writeDescriptorSet{};
+    VkWriteDescriptorSetAccelerationStructureKHR writeDescriptorSetAccelerationStructure{};
+    if (pObj)
+    {
+
+        writeDescriptorSet.sType = VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+        if (pObj->GetImageInfoCount())
+        {
+            writeDescriptorSet.descriptorCount = (uint32_t)pObj->GetImageInfoCount();
```

Comment:
For converting, we prefer C++ style, like static_cast<uint32_t>.

Reply:
Done